webapp/views/App/views/Designer/TaxonomyList/Row/store/actions/useDelete.js

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+
+import { DialogConfirmActions, NotificationActions } from '@webapp/store/ui'
+import * as Taxonomy from '@core/survey/taxonomy'
+import * as TaxonomyActions from '@webapp/loggedin/surveyViews/taxonomy/actions'
+
+import { useI18n } from '@webapp/store/system'
+
+import { State } from '../state'
+
+export const useDelete = ({ setState }) => {
+  const i18n = useI18n()
+  const dispatch = useDispatch()
+
+  return useCallback(({ state }) => {
+    const taxonomy = State.getTaxonomy(state)
+
+    if (taxonomy.usedByNodeDefs) {
```

Comment:
where do you initialize the usedByNodeDefs?

Reply:
Thank  when i moved the things 've lost this part

```A.isEmpty(Survey.getNodeDefsByTaxonomyUuid(Taxonomy.getUuid(taxonomy))(survey))```

Reply:
before we did that:

 const taxonomies = R.pipe(	
    Survey.getTaxonomiesArray,	
    R.map((t) => ({	
      ...t,	
      usedByNodeDefs: !R.isEmpty(Survey.getNodeDefsByTaxonomyUuid(Taxonomy.getUuid(t))(survey)),	
    }))	
  )(survey)